AI-Social-Platform.Services.Data/PublicationService.cs

```diff
@@ -0,0 +1,162 @@
+﻿using AI_Social_Platform.Data;
+using AI_Social_Platform.Data.Models.Publication;
+using AI_Social_Platform.Services.Data.Interfaces;
+using AI_Social_Platform.Services.Data.Models.PublicationDtos;
+using Microsoft.EntityFrameworkCore;
+
+namespace AI_Social_Platform.Services.Data;
+
+public class PublicationService : IPublicationService
+{
+    private readonly ASPDbContext dbContext;
+
+    public PublicationService(ASPDbContext dbContext)
+    {
+        this.dbContext = dbContext;
+    }
+    public async Task<IEnumerable<PublicationDto>> GetPublicationsAsync()
+    {
+        var publications = await
+            dbContext.Publications
+                .Include(p => p.Comments)
```

Comment:
Should comments be included when publications list are fetched? For better performance - comments should be called separately.

Reply:
I think the issue is fixed after the new commits. 
I kept the include only on DeletePublication Method because the logic is to delete the comments too.